### PR TITLE
CEMS-2079 CEMS-2065 CEMS-2080 CEMS-2081 CEMS-2082 /docdata /doctype /render /template history

### PR DESCRIPTION
**Pull request by tempcke**
https://homeceu.atlassian.net/browse/CEMS-2079
https://homeceu.atlassian.net/browse/CEMS-2065
https://homeceu.atlassian.net/browse/CEMS-2080
https://homeceu.atlassian.net/browse/CEMS-2081
https://homeceu.atlassian.net/browse/CEMS-2082

GET /docdata/{type}/{key}/history fix
GET /template/{type}/{key}/history
GET /docdata/{id}
GET /docdata/{type}/{key}
GET /doctype
GET /render - render via query args, accepts combinations of type, templateId, dataId, templateKey, dataKey

### CODE REVIEW

File: api/DocData/GetDocDataById.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      $dataId = $args['dataId'];
+      $entity = $this->useCase->getById($dataId);
+      $responseData = ResponseHelper::docDataModel($entity);
+      return $response
+          ->withStatus(200)
+          ->withJson($responseData);
+    } catch (RecordNotFoundException $e) {
+      return $response->withStatus(404);
+    } catch (Exception $e) {
+      return $response->withStatus(500, __CLASS__." failure");
```
Comment (ccopto) [inline api/DocData/GetDocDataById.php:39]: do you have a way to send it to a log?

Comment (tempcke) [reply api/DocData/GetDocDataById.php:39]: not yet, it is on my list to add monolog though

File: tests/Api/DocData/TestCase.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+
+namespace HomeCEU\Tests\Api\DocData;
+
+
+class TestCase extends \HomeCEU\Tests\Api\TestCase {
+
```
Comment (ccopto) [inline tests/Api/DocData/TestCase.php:8]: just in case you need it later?

**Review by ccopto — APPROVED** [review]

File: tests/Api/DocData/TestCase.php (hunk 2 above)
Comment (tempcke) [reply tests/Api/DocData/TestCase.php:8]: lol, fairly sure I had something in here, then perhaps I moved it out, hard to say.

**Pull request merged**